File: SDKs/beezup.com/2.0/go/pkg/models/shared/catalogcolumn.go
package shared

// CatalogColumn
// The catalog column configuration
type CatalogColumn struct {
	CatalogColumnName                  string                              `json:"catalogColumnName"`
	Configuration                      ColumnConfiguration                 `json:"configuration"`
	DuplicateProductValueConfiguration *DuplicateProductValueConfiguration `json:"duplicateProductValueConfiguration,omitempty"`
	ID                                 string                              `json:"id"`
	Ignored                            *bool                               `json:"ignored,omitempty"`
	Links                              CatalogColumnLinks                  `json:"links"`
	UserColumName                      string                              `json:"userColumName"`
}

File: SDKs/beezup.com/2.0/typescript/src/sdk/models/shared/catalogcolumn.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ColumnConfiguration } from "./columnconfiguration";
import { DuplicateProductValueConfiguration } from "./duplicateproductvalueconfiguration";
import { CatalogColumnLinks } from "./catalogcolumnlinks";



// CatalogColumn
/** 
 * The catalog column configuration
**/
export class CatalogColumn extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=catalogColumnName" })
  catalogColumnName: string;

  @SpeakeasyMetadata({ data: "json, name=configuration" })
  configuration: ColumnConfiguration;

  @SpeakeasyMetadata({ data: "json, name=duplicateProductValueConfiguration" })
  duplicateProductValueConfiguration?: DuplicateProductValueConfiguration;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=ignored" })
  ignored?: boolean;

  @SpeakeasyMetadata({ data: "json, name=links" })
  links: CatalogColumnLinks;

  @SpeakeasyMetadata({ data: "json, name=userColumName" })
  userColumName: string;
}
